### PR TITLE
Split up Satan's if statement

### PR DESCRIPTION
**Pull request by AndrewCarterUK**

### CODE REVIEW

**Comment by AndrewCarterUK** [comment]
Testing to see if an already defined `Error` class implemented `Throwable` was unnecessary inside this block:

``` php
if (!class_exists('ArithmeticError') || !class_exists('DivisionByZeroError')) {
    // ...
}
```

That code could only be run if (and only if) we were not in PHP 7 - so `Throwable` would either not exist or be user defined (and thus not actually mean throwable).

Also, `instanceof` should be used for testing if `Error` extends `Exception` - as `class_implements()` only returns implementations and not parent class names.

File: lib/intdiv.php [hunk 1]
```diff
@@ -28,34 +28,41 @@ function intdiv($dividend, $divisor)
...
+        if (!class_exists('ArithmeticError')) {
+            class ArithmeticError extends Error
             {
-                return $this->getMessage();
+                public function __toString()
+                {
+                    return $this->getMessage();
+                }
             }
+        } elseif (!ArithmeticError instanceof Exception) {
```
Comment (michaelcullum) [inline lib/intdiv.php:52]: If on PHP 7 then ArithmeticError is not an instance of Exception

File: lib/intdiv.php [hunk 2]
```diff
@@ -28,34 +28,41 @@ function intdiv($dividend, $divisor)
...
+        if (!class_exists('Error')) {
+            class Error extends Exception
             {
-                return $this->getMessage();
+                public function __toString()
+                {
+                    return $this->getMessage();
+                }
             }
+        } elseif (!Error instanceof Exception) {
```
Comment (michaelcullum) [inline lib/intdiv.php:40]: or implements throwable. You have to consider other polyfills.

File: lib/intdiv.php (hunk 1 above)
Comment (AndrewCarterUK) [inline lib/intdiv.php:52]: If on PHP 7, ArithmeticError and DivisionByZeroError are already defined, thus this code isn't executed

File: lib/intdiv.php (hunk 2 above)
Comment (AndrewCarterUK) [inline lib/intdiv.php:40]: Read above

File: lib/intdiv.php (hunk 1 above)
Comment (michaelcullum) [inline lib/intdiv.php:52]: This is a elseif...

**Pull request merged**